File: Largest_number.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        sort(nums.rbegin(),nums.rend(),[](auto a, auto b){
           if(to_string(a).size() ==to_string(b).size()){
               return b>a;
           }
       
           else{
               return to_string(b)+to_string(a)>to_string(a)+to_string(b) ;
               }
             });

             string ans;
    for(auto v : nums){
        ans+=to_string(v );
    }
    return (ans[0]=='0')?"0":ans ;
    }
   
};

File: main-One/largest-number.py
class Solution:
    def largestNumber(self, nums: List[int]) -> str:


        def comparator(x,y):
            p1,p2  =  str(x)+str(y),str(y)+str(x)
            if p1>p2:
                return -1
            else:
                return 0
            

        nums = sorted(nums,key=cmp_to_key(comparator))
        ans  = "".join([str(n) for n in nums])

        return "0" if int(ans)==0 else ans




